lib/web/apiserver.go

```diff
@@ -4118,6 +4239,68 @@ func contextWithMFAResponseFromRequestHeader(ctx context.Context, requestHeader
 	return ctx, nil
 }
 
+type wsBearerToken struct {
+	Token string `json:"token"`
+}
+
+type wsStatus struct {
+	Type    string `json:"type"`
+	Status  string `json:"status"`
+	Message string `json:"message,omitempty"`
+}
+
+var wsIODeadline = time.Second * 4
+
+// AuthenticateRequest authenticates request using combination of a session cookie
+// and bearer token retrieved from a websocket
+func (h *Handler) AuthenticateRequestWS(w http.ResponseWriter, r *http.Request) (*SessionContext, *websocket.Conn, error) {
```

Comment:
I notice that the corollary `AuthenticateRequest` method has these lines:

https://github.com/gravitational/teleport/blob/f8ad58c7c0194029aba1ea3ad1187dc382558f5f/lib/web/apiserver.go#L4200-L4202

which I don't see replicated anywhere in here. That said, I tested this with mfa required and everything worked as planned. Do you know why that's not necessary in this path?

Reply:
I dont know why it works when not included, these lines were added after i started the PR so I never noticed it, i think it makes sense to include them